### PR TITLE
feat: Use bgtask API to track progress of vfolder clone asynchronously

### PR DESCRIPTION
**Pull request by vesselofgod**
resolved : https://github.com/lablup/backend.ai/issues/354

**prevent blocking of the whole manager while cloning a vfolder**
- **Run the clone operation as a background task and just return the bgtask ID in the vfolder clone API, so that clients could keep track of the progress and result.**
clone pr

- **Split the destination vfolder creation and actual clone operations, and insert the database transaction to create the destination vfolder record between them.**
split

### CODE REVIEW

File: src/ai/backend/client/cli/vfolder.py [hunk 1]
```diff
@@ -556,19 +559,45 @@ def clone(name, target_name, target_host, usage_mode, permission):
             vfolder_info = session.VFolder(name).info()
             if not vfolder_info['cloneable']:
                 print("Clone is not allowed for this virtual folder. "
-                      "Please update the 'cloneable' option.")
+                    "Please update the 'cloneable' option.")
```
Comment (achimnol) [inline src/ai/backend/client/cli/vfolder.py:562]: Please fix the indent.

File: src/ai/backend/client/cli/vfolder.py [hunk 2]
```diff
@@ -556,19 +559,45 @@ def clone(name, target_name, target_host, usage_mode, permission):
...
+            bgtask_id = result['bgtask_id']
         except Exception as e:
             print_error(e)
             sys.exit(1)
 
+    async def clone_vfolder_tracker(bgtask_id):
+        async with AsyncSession() as session:
+            try:
+                bgtask = session.BackgroundTask(bgtask_id)
+                completion_msg_func = lambda: print_done("Vfolder cloned.")
```
Comment (achimnol) [inline src/ai/backend/client/cli/vfolder.py:579]: Let's write vfolders as "VFolder" or "vfolder" consistently, not "Vfolder".

File: src/ai/backend/client/cli/vfolder.py [hunk 3]
```diff
@@ -556,19 +559,45 @@ def clone(name, target_name, target_host, usage_mode, permission):
...
+                                pbar.write(data['message'])
+                                pbar.update(data['current_progress'] - pbar.n)
+                            elif ev.event == 'bgtask_failed':
+                                error_msg = data['message']
+                                completion_msg_func = \
+                                    lambda: print_fail(f"Error occurred: {error_msg}")
+                            elif ev.event == 'bgtask_cancelled':
+                                completion_msg_func = \
+                                    lambda: print_warn("Vfolder cloning has been "
+                                                    "cancelled in the middle.")
```
Comment (achimnol) [inline src/ai/backend/client/cli/vfolder.py:595]: Please fix the indent.

File: src/ai/backend/client/cli/vfolder.py [hunk 4]
```diff
@@ -2,17 +2,20 @@
 import json
 from pathlib import Path
 import sys
+from tqdm import tqdm
```
Comment (achimnol) [inline src/ai/backend/client/cli/vfolder.py:5]: Reorder the imports consistently.
(Q: What is the rule?)

**Review by achimnol — CHANGES_REQUESTED** [review]

File: src/ai/backend/client/cli/vfolder.py (hunk 4 above)
Comment (vesselofgod) [reply src/ai/backend/client/cli/vfolder.py:5]: Should I reorder it in alphabetical order?

**Pull request merged**